nima/sse/webclient/src/main/java/io/helidon/nima/sse/webclient/SseSourceHandlerProvider.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.sse.webclient;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+
+import io.helidon.common.GenericType;
+import io.helidon.common.media.type.MediaTypes;
+import io.helidon.nima.sse.SseEvent;
+import io.helidon.nima.webclient.ClientResponse;
+import io.helidon.nima.webclient.http.spi.Source;
+import io.helidon.nima.webclient.http.spi.SourceHandlerProvider;
+
+/**
+ * A handler for SSE sources.
+ */
+public class SseSourceHandlerProvider implements SourceHandlerProvider<SseEvent> {
+
+    private static final char BOM = 0xFEFF;
+    private static final String ID = "id:";
+    private static final String DATA = "data:";
+    private static final String RETRY = "retry:";
+    private static final String EVENT = "event:";
+
+    @Override
+    public boolean supports(GenericType<? extends Source<?>> type, ClientResponse response) {
+        return type == SseSource.TYPE && response.headers().contentType()
```

Comment:
I think we need to use `equals` here instead of `==`, because if you create a new `GenericType` with the same parameter, you should still pass this check.

Reply:
Done